src/main/java/name/webdizz/sonar/grammar/issue/tracking/LinkFunction.java

```diff
@@ -0,0 +1,78 @@
+package name.webdizz.sonar.grammar.issue.tracking;
+
+import com.google.common.base.Preconditions;
+import name.webdizz.sonar.grammar.GrammarPlugin;
+import name.webdizz.sonar.grammar.PluginParameter;
+import org.apache.commons.codec.binary.Base64;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.api.ServerExtension;
+import org.sonar.api.config.Settings;
+import org.sonar.api.issue.action.Function;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.apache.commons.lang.StringUtils.stripToEmpty;
+
+public class LinkFunction implements Function, ServerExtension {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LinkFunction.class);
+
+    @Override
+    public void execute(Context context) {
+        String word = getMistakeWord(context.issue().message(), PluginParameter.ERROR_DESCRIPTION);
+        LOGGER.info("try to write:" + word);
+        Settings settings = context.projectSettings();
+        String dict = stripToEmpty(settings.getString(PluginParameter.MANUAL_DICT));
+        saveProperties(dict + "," + word, PluginParameter.MANUAL_DICT, context);
+    }
+
+    private String getMistakeWord(String message, String errDescription) {
+        String res = message.replaceFirst(errDescription, "");
+        return res.substring(0, res.length() - 1);
+    }
+
+    private void saveProperties(String word, String dictName, Context context) {
```

Comment:
As far as I understand  from this thread  http://sonarqube.15.x6.nabble.com/sonar-dev-Using-settings-to-store-a-hidden-property-td5023322.html  there is no way to persist property  to db from pojo. We can read it and even write, but after restart server it disappear.

On the other hand we have "Web Service Java client" but regard  this post http://sonarqube.15.x6.nabble.com/sonar-dev-Web-Service-Java-Client-usage-tp4528518p4528523.html
we also cannot use it for persist data.


So, if you know best  way to do this  it  please inform me.